ext/serializer.c

```diff
@@ -966,6 +976,10 @@ void ddtrace_set_root_span_properties(ddtrace_root_span_data *span) {
             ZVAL_DOUBLE(&sample_rate, web_integration->get_sample_rate());
             zend_hash_str_add_new(metrics, ZEND_STRL("_dd1.sr.eausr"), &sample_rate);
         }
+
+        if (get_DD_TRACE_GIT_METADATA_ENABLED() && !is_opcache_preload()) {
```

Comment:
The [opcache preload test](https://github.com/DataDog/dd-trace-php/blob/44a07ba98de2062986a33cc1592a2b87533441a0/tests/opcache/preload.phpt#L4) was otherwise having a memory leak.

The leaked memory was the allocated memory by the cached zend strings initialized during the inject_from_git_dir procedure.

Why was it leaking?
The order of events was:
- The request is initialized
- The git metadata procedure is made and the strings allocated and cached
- _Something happens, resetting the content of the git_metadata global_
  - A root span is generated (cli.command), but it is not flushed because it doesn't get serialized since `DDTRACE_G(top_closed_stack)` is `NULL` during `ddtrace_serialize_closed_spans_with_cycle` 
- The request is re-initialized
- The git metadata procedure is made, and other strings are allocated and cached
- The request is made and spans serialized
- The git_metadata structure is cleaned
==> The "first" allocated git_metadata content wasn't freed.

From my point of view, I had two choices:
- Ensure the memory is freed so that we don't have a leak
- Don't inject the git metadata, since it anyway won't get serialized

Considering I couldn't find out how to do the first option, I defaulted to the second one.

Reply:
Okay thanks, I'll check this out later.